compiler/executable/function.rs

```diff
@@ -57,30 +61,152 @@ pub enum FunctionTablingType {
     Untabled,
 }
 
+pub trait FunctionCallCostProvider {
+    fn get_call_cost(&self, function_id: &FunctionID) -> Cost;
+}
+
+struct FunctionPlanner<'a, FIDType: FunctionIDAPI> {
+    cached_plans: &'a ExecutableFunctionRegistry,
```

Comment:
Here's a place where there's an exception to the rule, since the `ExecutableFunctionRegistry` has sibling types that overlap, so we can use naming to differentiate what they are doing.